commands/history/export.go

```diff
@@ -62,6 +63,22 @@ func runExport(ctx context.Context, dockerCli command.Cli, opts exportOptions) e
 			return errors.Errorf("no record found for ref %q", ref)
 		}
 
+		if opts.finalize {
+			for _, rec := range recs {
+				if rec.Trace == nil {
+					if err := finalizeRecord(ctx, rec.Ref, nodes); err != nil {
+						return err
+					}
+				}
+			}
+			recs, err = queryRecords(ctx, ref, nodes, &queryOptions{
```

Comment:
only do this if some records were finalized

Reply:
:see_no_evil: